gateway-service-impl/src/main/java/org/hypertrace/gateway/service/span/SpanService.java

```diff
@@ -185,46 +185,50 @@ private QueryRequest.Builder createQueryWithFilter(
   }
 
   private int getTotalFilteredSpans(RequestContext context, SpansRequest request) {
```

Comment:
This could return Optional<Integer> and let the caller decide whether to return -1 or something else.

Reply:
Not needed with new api def.